src/senselab/audio/tasks/preprocessing/preprocessing.py

```diff
@@ -103,40 +103,111 @@ def select_channel_from_audios(audios: List[Audio], channel_index: int) -> List[
     """
-    chunked_audios = []
-
+    extracted_segments = []
     for audio, timestamps in data:
-        start, end = timestamps
-        if start < 0:
-            raise ValueError("Start time must be greater than or equal to 0.")
-        duration = audio.waveform.shape[1] / audio.sampling_rate
-        if end > duration:
-            raise ValueError(f"End time must be less than the duration of the audio file ({duration} seconds).")
-        start_sample = int(start * audio.sampling_rate)
-        end_sample = int(end * audio.sampling_rate)
-        chunked_waveform = audio.waveform[:, start_sample:end_sample]
-
-        chunked_audios.append(
-            Audio(
-                waveform=chunked_waveform,
-                sampling_rate=audio.sampling_rate,
-                metadata=audio.metadata.copy(),
-                orig_path_or_id=audio.orig_path_or_id,
+        single_audio_segments = []
+        for start, end in timestamps:
+            if start < 0:
+                raise ValueError("Start time must be greater than or equal to 0.")
+            duration = audio.waveform.shape[1] / audio.sampling_rate
+            if end > duration:
+                raise ValueError(f"End time must be less than the duration of the audio file ({duration} seconds).")
+            start_sample = int(start * audio.sampling_rate)
+            end_sample = int(end * audio.sampling_rate)
+            chunked_waveform = audio.waveform[:, start_sample:end_sample]
+
+            single_audio_segments.append(
+                Audio(
+                    waveform=chunked_waveform,
+                    sampling_rate=audio.sampling_rate,
+                    metadata=audio.metadata.copy(),
+                    orig_path_or_id=audio.orig_path_or_id,
+                )
             )
+        extracted_segments.append(single_audio_segments)
+    return extracted_segments
+
+
+def pad_audios(audios: List[Audio], desired_samples: int) -> List[Audio]:
+    """Pads the audio segment to the desired length.
+
+    Args:
+        audios: The list of audio objects to be padded.
+        desired_samples: The desired length (in samples) for the padded audio.
+
+    Returns:
+        A new Audio object with the waveform padded to the desired length.
+    """
+    padded_audios = []
+    for audio in audios:
+        current_samples = audio.waveform.shape[1]
+
+        if current_samples >= desired_samples:
+            return [audio]
+
+        padding_needed = desired_samples - current_samples
+        padded_waveform = torch.nn.functional.pad(audio.waveform, (0, padding_needed))
+        padded_audio = Audio(
+            waveform=padded_waveform,
+            sampling_rate=audio.sampling_rate,
+            metadata=audio.metadata.copy(),
+            orig_path_or_id=audio.orig_path_or_id,
         )
-    return chunked_audios
+        padded_audios.append(padded_audio)
+    return padded_audios
+
+
+def evenly_segment_audios(
```

Comment:
same as above. not sure if people will actually need this functionality, but ok

Reply:
Yes, it works for stereo. I updated the test to check for this. I included this functionality because Summer School students were doing this with the B2AI audio files.